auto_install/install.sh

```diff
@@ -997,94 +1017,151 @@ installPiVPN(){
+			VPNChooseOptions=(WireGuard "" on
+								OpenVPN "" off)
+
+			if VPN=$("${chooseVPNCmd[@]}" "${VPNChooseOptions[@]}" 2>&1 >/dev/tty) ; then
+				echo "::: Using VPN: $VPN"
+				VPN="${VPN,,}"
+			else
+				echo "::: Cancel selected, exiting...."
+				exit 1
+			fi
+		elif [ "$X86_SYSTEM" -eq 0 ]; then
+			echo "::: Using VPN: OpenVPN"
+			VPN="openvpn"
 		fi
 	fi
 
-	if [ "$VPN" = "wireguard" ]; then
-		# Since WireGuard only uses UDP, askCustomProto() is never called so we
-		# set the protocol here (it's not actually required to save the value, but
-		# it might be useful for the user when port forwarding).
-		pivpnPROTO="udp"
-		echo "pivpnPROTO=${pivpnPROTO}" >> /tmp/setupVars.conf
-		pivpnDEV="wg0"
-		pivpnNET="10.6.0.0"
-	elif [ "$VPN" = "openvpn" ]; then
-		pivpnDEV="tun0"
-		pivpnNET="10.8.0.0"
+	echo "VPN=${VPN}" >> /tmp/setupVars.conf
+}
+
+downloadVerifyKey(){
+	local KEY_URL="$1"
+	local EXPECTED_KEY_ID="$2"
+
+	local KEY_CONTENT
+	local KEY_INFO
+	local DOWNLOADED_KEY_ID
+
+	if ! KEY_CONTENT="$(wget -qO- "$KEY_URL")"; then
+		return 1
 	fi
-	vpnGw="${pivpnNET/.0.0/.0.1}"
 
-	echo "VPN=${VPN}" >> /tmp/setupVars.conf
-	echo "pivpnDEV=${pivpnDEV}" >> /tmp/setupVars.conf
-	echo "pivpnNET=${pivpnNET}" >> /tmp/setupVars.conf
-	echo "subnetClass=${subnetClass}" >> /tmp/setupVars.conf
+	if ! KEY_INFO="$(gpg --show-key --with-colons <<< "$KEY_CONTENT")"; then
+		return 1
+	fi
+
+	DOWNLOADED_KEY_ID="$(sed -n '/^pub:/,/^fpr:/p' <<< "$KEY_INFO" | grep '^fpr' | cut -d ':' -f 10)"
+
+	if [ "$DOWNLOADED_KEY_ID" != "$EXPECTED_KEY_ID" ]; then
+		return 1
+	fi
+
+	echo "$KEY_CONTENT"
+	return 0
 }
 
 installOpenVPN(){
 	local PIVPN_DEPS
 
 	echo "::: Installing OpenVPN from Debian package... "
 
-	if [ "$PLAT" = "Debian" ] || [ "$PLAT" = "Ubuntu" ]; then
-		# gnupg is used by apt-key to import the openvpn GPG key into the
-		# APT keyring
-		PIVPN_DEPS=(gnupg)
-		installDependentPackages PIVPN_DEPS[@]
+	# Use x86-only OpenVPN APT repo on x86 Debian/Ubuntu systems
+	if [ "$PLAT" != "Raspbian" ] && [ "$X86_SYSTEM" -eq 1 ]; then
 
-		# We will download the repository key regardless of whether the user
-		# has already enabled the openvpn repository or not, just to make sure
-		# we have the right key
-		echo "::: Adding repository key..."
-		if ! $SUDO apt-key adv --keyserver keyserver.ubuntu.com --recv-keys "$OPENVPN_REPO_KEY"; then
-			echo "::: Failed to import OpenVPN GPG key"
-			exit 1
-		fi
+		AVAILABLE_OPENVPN="$(apt-cache policy openvpn | grep 'Candidate:' | awk '{print $2}')"
```

Comment:
The above check can lead to wrong results. See: https://github.com/pivpn/pivpn/commit/0a30365d65dc307781255953218085c4b864089d#r37792155

`apt-cache policy` will show `Candidate: (none)` if the package is available in repos but has been blocked from installed via APT preferences/priority pins.

Here it does not matter since `dpkg --compare-versions (none) ge 2.4` fails (with ugly error message about wrong syntax:
```
2020-03-16 16:02:48 root@micha:/tmp# dpkg --compare-versions "(none)" ge 2.4
dpkg: warning: version '(none)' has bad syntax: version number does not start with digit
```
However it is a bid dangerous probably to have a value assigned here? The failsafe alternative would be:
```
apt-cache policy openvpn | grep -m1 'Candidate: ' | grep -v '(none)' | awk '{print $2}
```
This way the variable stays empty if priority pins do not allow to install it, e.g. in case someone has added the unstable/sid repo but allowed only one/a few specific packages to be installed from it. In such case, before pulling a bunch of sid packages for dependencies, it is indeed better to apply the 3rd party repo.

Below there is a case where the apt-cache output is used as result directly...

Reply:
> apt-cache policy will show Candidate: (none) if the package is available in repos but has been blocked from installed via APT preferences/priority pins.

I forgot about that, thanks.

> apt-cache policy openvpn | grep -m1 'Candidate: ' | grep -v '(none)' | awk '{print $2}'

I think this also applies to: https://github.com/pivpn/pivpn/blob/6653d4caa31a5714971e03ae257c6ac9f7ebf206/auto_install/install.sh#L1187-L1188

Reply:
@orazioedoardo 
Yes, I edited that inside my post above. At best rule out "(none)" here as well with the same method and do not call `dpkg --compare-versions` if not both vars have a value (to avoid the error output).